app/uk/gov/hmrc/personaldetailsvalidation/model/PersonalDetails.scala

```diff
@@ -46,16 +59,49 @@ object PersonalDetails {
   }
 }
 
+
 case class PersonalDetailsWithNino(firstName: NonEmptyString,
                                    lastName: NonEmptyString,
                                    dateOfBirth: LocalDate,
                                    nino: Nino) extends PersonalDetails with PersonalDetailsNino {
+
+  override def addGender(gender: NonEmptyString): PersonalDetails =
+    PersonalDetailsWithNinoAndGender(firstName, lastName, dateOfBirth, nino, gender)
+
+  override def maybeNino: Option[Nino] = Some(nino)
```

Comment:
Seems a bit odd to have a maybeNino method in a class called "PersonalDetailsWithNino", but then I'm not really following all the changes clearly, sorry.